craft_application/models/project.py

```diff
@@ -73,6 +73,8 @@ class Project(CraftBaseModel):
 
     parts: Dict[str, Dict[str, Any]]  # parts are handled by craft-parts
 
+    package_repositories: Optional[List[Dict[str, Any]]]
```

Comment:
Is there a reason for declaring this as such instead of using the package repository type? This would also allow us to have automatic validation instead of doing it by hand.

Reply:
I don't see any reason why not